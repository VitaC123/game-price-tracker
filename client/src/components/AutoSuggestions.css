.suggestion {
    max-width: 50vw;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}