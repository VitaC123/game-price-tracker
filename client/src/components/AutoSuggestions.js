import React, { Component } from 'react';
import { browserHistory } from 'react-router'
import { Badge, InputGroupButton, Dropdown, DropdownMenu, DropdownItem } from 'reactstrap';


function SuggestedResult(props) {
    const { title, image } = props;
    const { platforms } = props.details;

    return (
        <div className='suggestion'>
            {platforms.map((system, i) => <span key={system}><Badge pill>{system}</Badge>{' '}</span>)}
            <small>{title}</small>
            <img className='suggestionImage' src={image} alt={title + ' cover'} />
        </div>
    );
}

class AutoSuggestions extends Component {
    constructor(props) {
        super(props);
        this.state = { toggle: false };
        this.toggle = this.toggle.bind(this);
    }
    toggle() {
        this.setState({ toggle: !this.state.toggle });
    }
    handleClick(gameId) {
        browserHistory.push(`/games/${gameId}`);
    }
    render() {
        const suggestions = this.props.suggestions.slice(0, 5);

        return (
            <InputGroupButton>
                <Dropdown isOpen={suggestions.length > 0} toggle={this.toggle}>
                    <DropdownMenu>
                        {suggestions.map((game, i) => {
                            return (
                                <div key={game.cid + '-' + i}>
                                    <DropdownItem onClick={() => this.handleClick(game._id)}>
                                        <SuggestedResult {...game} />
                                    </DropdownItem>
                                    <DropdownItem divider />
                                </div>
                            );
                        })}
                    </DropdownMenu>
                </Dropdown>
            </InputGroupButton>
        );
    }
}

export default AutoSuggestions;
