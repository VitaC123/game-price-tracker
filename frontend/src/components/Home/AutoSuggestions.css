.suggestion {
  width: 50vw;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.autoSuggestionsDropdownToggle {
  visibility: hidden;
  padding: 0;
}
