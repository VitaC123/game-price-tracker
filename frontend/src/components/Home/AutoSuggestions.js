import React, { PureComponent } from 'react';
import { push } from 'react-router-redux';
import store from '../../redux/store';

import { Badge, InputGroupButton, Dropdown, DropdownMenu, DropdownItem, DropdownToggle } from 'reactstrap';
import './AutoSuggestions.css';


const Suggestion = props => {
    const { title, image } = props;
    const { platforms } = props.details;
    return (
        <div className='suggestion'>
            {platforms.map(system => <span key={system}><Badge pill>{system}</Badge>{' '}</span>)}
            <small>{title}</small>
            <img className='suggestionImage' src={image} alt={title} />
        </div>
    );
};

class AutoSuggestions extends PureComponent {
    constructor(props) {
        super(props);
        this.toggle = this.toggle.bind(this);
        this.state = { dropdownOpen: false };
    }
    toggle() {
        this.setState({ dropdownOpen: !this.state.dropdownOpen });
    }
    handleClick(gameId) {
        store.dispatch(push(`/games/${gameId}`));
    }
    render() {
        const autoSuggestions = this.props.autoSuggestions ? this.props.autoSuggestions : [];
        return (
            <InputGroupButton>
                <Dropdown isOpen={autoSuggestions.length > 0} toggle={this.toggle}>
                    <DropdownToggle className='autoSuggestionsDropdownToggle' />
                    <DropdownMenu>
                        {autoSuggestions.map((game, i) => {
                            return (
                                <div key={game.cid + '-' + i}>
                                    <DropdownItem onClick={() => this.handleClick(game._id)}>
                                        <Suggestion {...game} />
                                    </DropdownItem>
                                    <DropdownItem divider />
                                </div>
                            );
                        })}
                    </DropdownMenu>
                </Dropdown>
            </InputGroupButton>
        );
    }
}


export default AutoSuggestions;
