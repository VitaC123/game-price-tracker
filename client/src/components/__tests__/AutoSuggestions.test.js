import React from 'react';
import AutoSuggestions from '../AutoSuggestions';
import renderer from 'react-test-renderer';


const stubData = [{
  _id: 'UP9000-CUSA00553_00-THELASTOFUS00000',
  title: 'The Last of Us: Left Behind Stand Alone',
  url: 'https://store.playstation.com/#!/en-us/games/cid=UP9000-CUSA00553_00-THELASTOFUS00000',
  price: '9.99',
  strikePrice: null,
  onSale: false,
  discount: null,
  psPlusPrice: null,
  lastUpdated: 1501570800000,
  image: 'https://store.playstation.com/store/api/chihiro/00_09_000/container/US/en/19/UP9000-CUSA00553_00-THELASTOFUS00000/image?w=225&h=225',
  details: {
    platforms: ['PS4â„¢']
  }
}];

it('renders correctly', () => {
  const tree = renderer.create(
    <AutoSuggestions autoSuggestions={stubData} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
