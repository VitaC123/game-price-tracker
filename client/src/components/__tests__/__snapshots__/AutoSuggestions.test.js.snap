// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="input-group-btn"
>
  <div
    className="dropdown show"
  >
    <div
      aria-hidden={false}
      className="dropdown-menu"
      role="menu"
      tabIndex="-1"
    >
      <div>
        <button
          className="dropdown-item"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          <div
            className="suggestion"
          >
            <span>
              <span
                className="badge badge-default badge-pill"
              >
                PS4â„¢
              </span>
               
            </span>
            <small>
              The Last of Us: Left Behind Stand Alone
            </small>
            <img
              alt="The Last of Us: Left Behind Stand Alone cover"
              className="suggestionImage"
              src="https://store.playstation.com/store/api/chihiro/00_09_000/container/US/en/19/UP9000-CUSA00553_00-THELASTOFUS00000/image?w=225&h=225"
            />
          </div>
        </button>
        <div
          className="dropdown-divider"
          onClick={[Function]}
          tabIndex="-1"
          type={undefined}
        />
      </div>
    </div>
  </div>
</div>
`;
